java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -90,9 +99,9 @@ abstract class AbstractArcHost(vararg initialParticles: ParticleRegistration) :
     protected fun getArcHostContext(arcId: String) = contextCache[arcId]
 
     protected suspend fun lookupOrCreateArcHostContext(
-        partition: Plan.Partition
-    ): ArcHostContext = contextCache[partition.arcId] ?: readContextFromStorage(
-        createArcHostContext(partition.arcId)
+        arcId: String
```

Comment:
Someday perhaps we'd benefit from having arcId be an actual type rather than just a string? To improve validation, log formatting, inherent documentation, etc. To consider for future.

Reply:
There's already an `ArcId` type. I had attempted a PR to change everything over, but there was a lot of churn underneath, so I decided to hold off so as not to introduce further distractions. 

Maybe it would be better to do the `String`->`ArcId` conversation as a series of small PRs.

Reply:
Yeah. Let's do it post-march cleanup.